donate/payments/views.py

```diff
@@ -314,27 +314,35 @@ def process_monthly_transaction(self, form):
                     'eventLabel': 'Monthly',
                 }
             ])
-            return self.success(
-                result,
-                form,
-                transaction_id=result.subscription.id,
-                last_4=payment_method.last_4,
-                send_data_to_basket=False,
-            )
+            # Bypass self.success, as this is not a transaction but a subscription:
```

Comment:
Monthly does not involve a transaction, so falling through to success() which then falls through to getting the transaction details, isn't actually required. We can immediately redirect to the success URL

Reply:
The more I think about it, this feels like setting us up to missing something in the future. I suspect we should either rename `success` to be something like `transactionSuccess` or check if this was indeed a transaction inside the success method.

I'm trying to think about all the different code pathways we have in this stack and keep them from branching out as much as possible/keep the flow clear.

Reply:
I'm in favour of renaming `.success` to a function that makes it explicit that this is only for transactions.